Compiler/NFFrontEnd/NFTyping.mo

```diff
@@ -445,6 +609,33 @@ algorithm
   end match;
 end makeTypeAttribute;
 
+function typeExpOption
+  input Option<Absyn.Exp> absOpExp;
+  input Component.Scope scope;
+  input ComponentNode component;
+  input SourceInfo info;
+  output Option<DAE.Exp> typedOpExp;
+  output Option<DAE.Type> opTy;
+  output Option<DAE.Const> OpVarity;
+protected
+  Absyn.Exp absExp;
+  DAE.Exp e1;
+  DAE.Type ty1;
+  DAE.Const var1;
+algorithm
+  if isSome(absOpExp) then
+    SOME(absExp) := absOpExp;
+    (e1, ty1, var1) := typeExp(absExp, NFComponent.DEFAULT_SCOPE, component, info);
+    typedOpExp := SOME(e1);
+    opTy := SOME(ty1);
+    OpVarity := SOME(var1);
+  else
+    typedOpExp := NONE();
+    opTy := NONE();
+    OpVarity := NONE();
+  end if;
+end typeExpOption;
```

Comment:
You don't need it if you match on SOME() / NONE() in the pattern for the for-loop. Then you can avoid creating 3 SOME-nodes.